foursight_core/routes.py

```diff
@@ -0,0 +1,239 @@
+        """
+        Redirect with 302 to view page of DEFAULT_ENV
+        Non-protected route
+        """
+        redirect_path = ROUTE_PREFIX_EXPLICIT + 'view/' + app.core.get_default_env()
+        headers = {'Location': redirect_path}
+        return Response(status_code=302, body=json.dumps(headers), headers=headers)
+
+    @staticmethod
+    @app.route(ROUTE_PREFIX + "view", methods=['GET'])
+    def route_view():
+        redirect_path = ROUTE_PREFIX_EXPLICIT + 'view/' + app.core.get_default_env()
+        headers = {"Location": redirect_path}
+        return Response(status_code=302, body=json.dumps(headers), headers=headers)
+
+    @staticmethod
+    @app.route(ROUTE_PREFIX + 'introspect', methods=['GET'])
+    def introspect(environ):
+        """
+        Test route
+        """
+        auth = app.core.check_authorization(app.current_request.to_dict(), environ)
+        if auth:
+            return Response(status_code=200, body=json.dumps(app.current_request.to_dict()))
+        else:
+            return app.core.forbidden_response()
+
+    @staticmethod
+    @app.route(ROUTE_PREFIX + 'view_run/{environ}/{check}/{method}', methods=['GET'])
+    def view_run_route(environ, check, method):
+        """
+        Protected route
+        """
+        req_dict = app.current_request.to_dict()
+        domain, context = app.core.get_domain_and_context(req_dict)
+        query_params = req_dict.get('query_params', {})
+        if app.core.check_authorization(req_dict, environ):
+            if method == 'action':
+                return app.core.view_run_action(environ, check, query_params, context)
+            else:
+                return app.core.view_run_check(environ, check, query_params, context)
+        else:
+            return app.core.forbidden_response(context)
+
+    @staticmethod
+    @app.route(ROUTE_PREFIX + 'view/{environ}', methods=['GET'])
+    def view_route(environ):
+        """
+        Non-protected route
+        """
+        req_dict = app.current_request.to_dict()
+        domain, context = app.core.get_domain_and_context(req_dict)
+        return app.core.view_foursight(app.current_request, environ, app.core.check_authorization(req_dict, environ), domain, context)
+
+    @staticmethod
+    @app.route(ROUTE_PREFIX + 'view/{environ}/{check}/{uuid}', methods=['GET'])
+    def view_check_route(environ, check, uuid):
+        """
+        Protected route
+        """
+        req_dict = app.current_request.to_dict()
+        domain, context = app.core.get_domain_and_context(req_dict)
+        if app.core.check_authorization(req_dict, environ):
+            return app.core.view_foursight_check(app.current_request, environ, check, uuid, True, domain, context)
+        else:
+            return app.core.forbidden_response()
+
+    @staticmethod
+    @app.route(ROUTE_PREFIX + 'history/{environ}/{check}', methods=['GET'])
+    def history_route(environ, check):
+        """
+        Non-protected route
```

Comment:
It's useful to know what's protected or not, but also say what the method does.

Also, how does the system know what's protected or not? Are only the `view` things protected?

Maybe make a ticket so that we make a decorator to help make this clearer.

Reply:
This is the non-React stuff which I tried to touch minimally.